docs/develop/python/message-passing.mdx

```diff
@@ -221,132 +247,214 @@ async def main():
 
-Workflow Updates handlers are methods in your Workflow Definition designed to handle updates.
-These updates can be triggered during the lifecycle of a Workflow Execution.
+```python
+@activity.defn
+async def call_greeting_service(to_language: Language) -> Optional[str]:
+    await asyncio.sleep(0.2)  # Pretend that we are calling a remote service.
+    greetings = {
+        Language.Arabic: "مرحبا بالعالم",
+        Language.Chinese: "你好，世界",
+        Language.English: "Hello, world",
+        Language.French: "Bonjour, monde",
+        Language.Hindi: "नमस्ते दुनिया",
+        Language.Spanish: "Hola mundo",
+    }
+    return greetings.get(to_language)
 
-**Define an Update Handler**
 
-To define an update handler, use the [@workflow.update](https://python.temporal.io/temporalio.workflow.html#update) decorator on a method within your Workflow. This decorator can be applied to both asynchronous and synchronous methods.
+@workflow.defn
+class GreetingWorkflow:
+    ...
+    @workflow.update
+    async def set_language(self, language: Language) -> Language:
+        if language not in self.greetings:
+            greeting = await workflow.execute_activity(
+                call_greeting_service,
+                language,
+                start_to_close_timeout=timedelta(seconds=10),
+            )
+            if greeting is None:
+                # An update validator cannot be async, so cannot be used to check that the remote
+                # call_greeting_service supports the requested language. Raising ApplicationError
+                # will fail the Update, but the WorkflowExecutionUpdateAccepted event will still be
+                # added to history.
+                raise ApplicationError(
+                    f"Greeting service does not support {language.name}"
+                )
+            self.greetings[language] = greeting
+        previous_language, self.language = self.language, language
+        return previous_language
+```
 
-- **Decorator Usage:** Apply `@workflow.update` to the method intended to handle updates.
-- **Overriding:** If a method with this decorator is overridden, the overriding method should also be decorated with `@workflow.update`.
-- **Validator Method:** Optionally, you can define a validator method for the update handler. This validator is specified using `@update_handler_method_name.validator` and is invoked before the update handler.
-- **Method Parameters:** Update handlers should only use positional parameters. For non-dynamic methods, it's recommended to use a single parameter that is an object or data class, which allows for future expansion of fields.
-- **Return Values:** The update handler can return a serializable value. This value is sent back to the caller of the update.
 
+#### Waiting
```

Comment:
This is such an important concept and `wait_condition` is such a powerful/common tool for a workflow, it may also deserve its own section unrelated to message. Doesn't affect this PR, but we often find people don't know that this tool exists.

Reply:
Thanks, yes I emphasized that `wait_condition` is very useful in order to try to avoid it being overlooked.